src/journal/journald-context.c

```diff
@@ -784,3 +840,40 @@ void client_context_acquire_default(Server *s) {
 
         }
 }
+
+#if HAVE_PCRE2
+static bool is_log_matched(pcre2_code *regex, const char *message) {
+        int r;
+        _cleanup_(sym_pcre2_match_data_freep) pcre2_match_data *md = NULL;
+
+        assert(regex);
+
+        md = sym_pcre2_match_data_create(1, NULL);
+        if (!md)
+                return log_oom();
```

Comment:
Can we cache this in ClientContext as well? It's kinda annoying that we can't pass this as NULL since we don't use the result.